metsim/metsim.py

```diff
@@ -690,4 +739,40 @@ def wrap_run(func: callable, loc: dict, params: dict,
 
     # Cut the returned data down to the correct time index
     df_complete = df_complete.loc[new_times[0]:new_times[-1]]
-    return (loc, df_complete, df_base)
+    return df_complete, df_base
+
+
+@dask.delayed()
+def wrap_run_slice(params, write_lock, domain_slice=NO_SLICE):
+    print('running chunk %s' % domain_slice, flush=True)
+    os.environ["HDF5_USE_FILE_LOCKING"] = HDF5_USE_FILE_LOCKING
+    ms = MetSim(params, domain_slice=domain_slice)
+    # TODO: determine if this lock/block is actually necessary.
+    # lock = SerializableLock()
+    # with lock:
+    #     print("load_inputs")
```

Comment:
After thinking about this a bit more, I think we should still protect with a threading lock here.